packages/rum-core/src/domain/action/getActionNameFromElement.ts

```diff
@@ -288,12 +279,15 @@ function getTextualContentWithTreeWalker(
   return text.replace(/\s+/g, ' ').trim()
 
   function rejectInvisibleOrMaskedElementsFilter(node: Node) {
+    const nodeSelfPrivacyLevel = getNodePrivacyLevel(node, nodePrivacyLevel, nodePrivacyLevelCache)
```

Comment:
Instead of passing in `nodePrivacyLevel` for the second argument of `getNodePrivacyLevel()`, you should be passing in the `defaultPrivacyLevel`. You can just replace the `nodePrivacyLevel` parameter to `getTextualContentWithTreeWalker()` with a `defaultPrivacyLevel` parameter, and then do:
```suggestion
    const nodeSelfPrivacyLevel = getNodePrivacyLevel(node, defaultPrivacyLevel, nodePrivacyLevelCache)
```
When calling `getTextualContentWithTreeWalker()`, you can get the default privacy level from the `RumConfiguration` via `configuration.defaultPrivacyLevel`.

(It'd also be fine to just pass the `RumConfiguration` directly to `getTextualContentWithTreeWalker()`, if you prefer.)

I think this means that you can also remove the `nodePrivacyLevel` parameter from a bunch of calling functions, but double check me on that.